Plugins/LfpDisplayNode/LfpDisplayNode.cpp

```diff
@@ -353,62 +359,68 @@ void LfpDisplayNode::handleEvent(const EventChannel* eventInfo, const MidiMessag
+        
+        if (nSamples < samplesLeft)
+        {
+
+            displayBuffers[i]->copyFrom (chan,                                      // destChannel
+                                     displayBufferIndices[i][chan],             // destStartSample
+                                     arrayOfOnes,                               // source
+                                     nSamples,                                  // numSamples
+                                     float (ttlState[subprocessorToDraw]));     // gain
+        }
+        else
+        {
+            int extraSamples = nSamples - samplesLeft;
+
+            displayBuffers[i]->copyFrom (chan,                                      // destChannel
+                                     displayBufferIndices[i][chan],             // destStartSample
+                                     arrayOfOnes,                               // source
+                                     samplesLeft,                               // numSamples
+                                     float (ttlState[subprocessorToDraw]));     // gain
+
+            displayBuffers[i]->copyFrom (chan,                                      // destChannel
+                                     0,                                         // destStartSample
+                                     arrayOfOnes,                               // source
+                                     extraSamples,                              // numSamples
+                                     float (ttlState[subprocessorToDraw]));     // gain
+        }
+    }
 }
 
 void LfpDisplayNode::finalizeEventChannels()
 {
-	for (int i = 0 ; i < numSubprocessors ; i++){	
-		const int chan          = numChannelsInSubprocessor[allSubprocessors[i]];
-		const int samplesLeft   = displayBuffers[i]->getNumSamples() - displayBufferIndices[i][chan];
-		const int nSamples      = getNumSourceSamples(allSubprocessors[i]);
+    for (int i = 0 ; i < numSubprocessors ; i++){    
+        const int chan          = numChannelsInSubprocessor[allSubprocessors[i]];
+        const int index = displayBufferIndices[i][chan];
+        const int samplesLeft   = displayBuffers[i]->getNumSamples() - index;
+        const int nSamples      = getNumSourceSamples(allSubprocessors[i]);
+        
+        int newIdx = 0;
         
-		int newIdx = 0;
+        if (nSamples < samplesLeft)
+        {
+            newIdx = displayBufferIndices[i][chan] + nSamples;
+        }
+        else
+        {
+            newIdx = nSamples - samplesLeft;
+        }
         
-		if (nSamples < samplesLeft)
-		{
-			newIdx = displayBufferIndices[i][chan] + nSamples;
-		}
-		else
-		{
-			newIdx = nSamples - samplesLeft;
-		}
+        displayBufferIndices[i][chan] = newIdx;
+        if (latestCurrentTrigger >= 0)
+        {
+            latestTrigger = latestCurrentTrigger + index;
+        }
+    }
```

Comment:
A major difference between "beta" LFPViewer and "main" LFPViewer is that the main one supports switching between different sub-processors while data acquisition is active. This means that, when there are multiple inputs connected to the main LFPViewer, one can switch between these inputs on the fly by selecting any one from the drop-down menu of LFPViewer in the signal chain. To provide this functionality, instead of having a single display buffer, LfpDisplayNode now manages multiple display buffers, one for each input sub-processor. 

Considering all of this, the way you update the `lastTrigger` value in the `finalizeEventChannhels()` method should be different as compared to the beta version. As you can see in the method definition, the main version has a for loop that iterates through each input sub-processor, and then updates the `displayBufferIndex` for that sub-processor's event channel with new index value. But, we don't need to update the `lastTrigger` value for each sub-processor, because if we do that for all sub-processors, then the `lastTrigger` value is always going to use the `displayBufferIndex` value of the last sub-processor in the list, even though we are drawing any other processor and not the last one. I tested your PR with a neuropixels module with 2 probes attached to it, and found out that this causes performance issues and CPU usage spikes every time an event-triggered display occurs. It also makes the GUI freeze sometimes, especially when the sub-processors you are switching between have different sample rates. I have recorded a video of my test, you can watch that video via this link : https://drive.google.com/file/d/1KjJPVZrE8Z5rNMPrxUTqiBx818zj3PFv/view

That is why we only need to set the `lastTrigger` value for the sub-processor we are currently drawing. We need to move the that segment of code outside the for loop. So, my suggested change looks something like this:

```suggestion
    }
    if (latestCurrentTrigger >= 0)
    {
        int chan = numChannelsInSubprocessor[subprocessorToDraw];
        int subProcIndex = allSubprocessors.indexOf(subprocessorToDraw);
        latestTrigger = latestCurrentTrigger + displayBufferIndices[subProcIndex][chan];
    }
```
With this change, I'm able to get rid of the performance issue I was facing earlier, and everything works fine. My question is, should I merge the PR as-is and then make my changes right after, or do you want to test this change first and then add a new commit to this PR?

Also, in LfpDisplayCanvas.cpp, you say that "trigger event is positioned at 1/3 of the display width", but that is not the case. In reality, the trigger event is positioned at 1/4 of display width. Can you please confirm this?